agents/agents-features/agents-features-snapshot/src/commonMain/kotlin/ai/koog/agents/snapshot/feature/AgentCheckpointFeatureConfig.kt

```diff
@@ -0,0 +1,36 @@
+package ai.koog.agents.snapshot.feature
+
+import ai.koog.agents.features.common.config.FeatureConfig
+import ai.koog.agents.snapshot.providers.NoAgentCheckpointStorageProvider
+import ai.koog.agents.snapshot.providers.AgentCheckpointStorageProvider
+
+
+/**
+ * Configuration class for the Snapshot feature.
+ */
+public class AgentCheckpointFeatureConfig: FeatureConfig() {
+
+    internal var agentCheckpointStorageProvider: AgentCheckpointStorageProvider = NoAgentCheckpointStorageProvider()
+        private set
+
+    internal var continuousPersistense: Boolean = false
+        private set
+
+    /** Sets the [AgentCheckpointStorageProvider] for this feature.
+     * @param agentCheckpointStorageProvider The provider to set.
+     */
+    public fun snapshotProvider(agentCheckpointStorageProvider: AgentCheckpointStorageProvider) {
+        this.agentCheckpointStorageProvider = agentCheckpointStorageProvider
+    }
```

Comment:
1. Let's mb call it `storageProvider` or even just `storage` ?
2. If it's not adding +1 another storage but actually re-setting the single `this.agentCheckpointStorageProvider` maybe let's make it just a field with public setter?

```
install(Checkpoints) {
  storageProvider = MyStorageProvider()
  
  enableAutomaticPersistency = true // or whatever name
}
```